game-core/src/test/java/games/strategy/triplea/TripleATest.java

```diff
@@ -0,0 +1,59 @@
+package games.strategy.triplea;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.chat.Chat;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.framework.ServerGame;
+import games.strategy.engine.framework.startup.launcher.LaunchAction;
+import games.strategy.engine.framework.startup.ui.PlayerType;
+import games.strategy.engine.player.Player;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TripleATest {
+
+  private static final String DELEGATE_NAME_EDIT = "edit";
+
+  @Mock private LaunchAction launchAction;
+  @Mock private Chat chat;
+  @Mock private ServerGame serverGame;
+  @Mock private GameData gameData;
+
+  @Test
+  void testStartGameWhenServerGameStartedWithNewPlayersWithoutEditDelegate() {
+
+    final TripleA tripleA = new TripleA();
+    final Map<String, PlayerType> playerNames = new HashMap<>();
+    playerNames.put("first", PlayerType.HUMAN_PLAYER);
+    playerNames.put("second", PlayerType.WEAK_AI);
+    playerNames.put("third", PlayerType.PRO_AI);
+
+    when(serverGame.getData()).thenReturn(gameData);
+    when(gameData.getDelegate(DELEGATE_NAME_EDIT)).thenReturn(null);
+
+    final Set<Player> players = tripleA.newPlayers(playerNames);
+    assertThat(players, hasSize(3));
+
+    tripleA.startGame(serverGame, players, launchAction, chat);
+    verify(gameData).addDelegate(any());
+    verify(serverGame).addDelegateMessenger(any());
+    verify(serverGame).setDisplay(any());
+    verify(serverGame).setSoundChannel(any());
+
+    tripleA.shutDown();
+    verify(serverGame, times(2)).setDisplay(any());
```

Comment:
The second time, we call `setDisplay`, we expect `null` to be passed, right?

Generally asserting `any()` in mockito is a test smell indicating a weak test. WDYT to either eliminate this verification or to verify we get an expected `null` value the second time? 

Did you find that the `setDisplay` to `null` was strictly necessary/important, or could we remove it sometime soon? There are a number of examples of unnecessary setting to null, essentially manual GC when we can let the Java GC just do its thing for us. It's hard to say for sure though as the state is finicky...

Reply:
I haven't found how to verify that the sound channel has been set to null.
Instead of any(), is it better to say `.setDisplay(isNull())` ?

I don't know why they are setting it to null BTW, I didn't write/edit the TripleaA class and thought it's best to leave it as is. :)